api/controllers/UserController.js

```diff
@@ -2,41 +2,61 @@ import jwt from 'jsonwebtoken'
 import { getPublicKeyFromPem } from '../../lib/util'
 
 module.exports = {
-  create: function (req, res) {
-    const { name, email, email_validated, password } = req.allParams()
 
-    return User.create({name, email: email ? email.toLowerCase() : null, email_validated, account: {type: 'password', password}})
-    .then(async (user) => {
-      await Analytics.trackSignup(user.id, req)
-      if (req.param('login')) {
-        await UserSession.login(req, user, 'password')
+  create: async function (req, res) {
+    const { name, email, groupId } = req.allParams()
+    const group = groupId && await Group.find(groupId)
+
+    let user = await User.find(email, {}, false)
+    if (user) {
+      // User already exists so just invite them to the new group (if there is one)
+      if (group) {
+        // TODO: create Invitation object and send, but from what user? or do we add them directly to the group? but then they start receiving emails without consenting...
+        return res.ok({})
+      } else {
+        // TODO: No group to add them to and they already exist, so dont do anything for now? return ok or error?
+        return res.status(422).send({ error: "duplicate-email" })
```

Comment:
Yeah, it does seem like just returning OK here is a good thing, maybe sending something back in the header to indicate that the membership already existed would be a bonus in case some UI was want of that information at some point.

Reply:
ok look at this again, see what i did